### PR TITLE
Fix programmatic chain switching issue in Magic connector

### PR DESCRIPTION
**Pull request by Adebesin-Cell**
# Fix programmatic chain switching issue in Magic connector

This PR addresses an issue where programmatic chain switching in the Magic connector was causing an error message indicating that Magic does not support this feature. The fix involves adjusting the chain switching logic to accommodate the requirements of Magic, ensuring smooth switching functionality without encountering errors.

Fixes: https://github.com/EveripediaNetwork/issues/issues/2630

### CODE REVIEW

File: src/lib/magic/connectors/dedicatedWalletConnector.ts [hunk 1]
```diff
@@ -245,5 +256,66 @@ export function dedicatedWalletConnector({
...
+
+            return chain
+          } catch (error) {
+            throw new UserRejectedRequestError(error as Error)
+          }
+        }
+
+        throw new SwitchChainError(error as Error)
+      }
+    },
```
Comment (Royal-lobster) [inline src/lib/magic/connectors/dedicatedWalletConnector.ts:319]: @Adebesin-Cell I am lost 😅 Where are we checking if its magic ? if so bypass the error ?

**Comment by Royal-lobster** [comment]
@Adebesin-Cell now with injected, its now switching

https://github.com/EveripediaNetwork/ep-ui/assets/52039218/888fd3bb-80ef-4b90-8500-282a40eb8645

**Comment by Adebesin-Cell** [comment]
> @Adebesin-Cell now with injected, its now switching
> 
>  CleanShot.2024-05-09.at.15.28.32.mp4

Hm, weird. I’ll check again, it’s not loading a particular effect that gets current chain that’s what I’m thinking. But it should :) I’ll check again to see why it fails. 

One major thing: We have too many patches atm to make things work :)) We should take time to refactor as well as deprecate old workarounds.

**Comment by Royal-lobster** [comment]
> > @Adebesin-Cell now with injected, its now switching
> > CleanShot.2024-05-09.at.15.28.32.mp4
> 
> Hm, weird. I’ll check again, it’s not loading a particular effect that gets current chain that’s what I’m thinking. But it should :) I’ll check again to see why it fails.
> 
> One major thing: We have too many patches atm to make things work :)) We should take time to refactor as well as deprecate old workarounds.

I Think its better we consolidate all these patches into one pr, we need to ensure everything works when they all are combined

**Comment by Softdev1** [comment]
> > > @Adebesin-Cell now with injected, its now switching
> > > CleanShot.2024-05-09.at.15.28.32.mp4
> > 
> > 
> > Hm, weird. I’ll check again, it’s not loading a particular effect that gets current chain that’s what I’m thinking. But it should :) I’ll check again to see why it fails.
> > One major thing: We have too many patches atm to make things work :)) We should take time to refactor as well as deprecate old workarounds.
> 
> I Think its better we consolidate all these patches into one pr, we need to ensure everything works when they all are combined

Yeah, the fix works for me, but I still feel there are some cases we should consider @Adebesin-Cell  as  the fix @Royal-lobster  isn't working for him. Please let's test well, please

**Comment by Adebesin-Cell** [comment]
> > > > @Adebesin-Cell now with injected, its now switching
> > > > CleanShot.2024-05-09.at.15.28.32.mp4
> > > 
> > > 
> > > Hm, weird. I’ll check again, it’s not loading a particular effect that gets current chain that’s what I’m thinking. But it should :) I’ll check again to see why it fails.
> > > One major thing: We have too many patches atm to make things work :)) We should take time to refactor as well as deprecate old workarounds.
> > 
> > 
> > I Think its better we consolidate all these patches into one pr, we need to ensure everything works when they all are combined
> 
> Yeah, the fix works for me, but I still feel there are some cases we should consider @Adebesin-Cell as the fix @Royal-lobster isn't working for him. Please let's test well, please

For sure!

**Comment by Adebesin-Cell** [comment]
> > > @Adebesin-Cell now with injected, its now switching
> > > CleanShot.2024-05-09.at.15.28.32.mp4
> > 
> > 
> > Hm, weird. I’ll check again, it’s not loading a particular effect that gets current chain that’s what I’m thinking. But it should :) I’ll check again to see why it fails.
> > One major thing: We have too many patches atm to make things work :)) We should take time to refactor as well as deprecate old workarounds.
> 
> I Think its better we consolidate all these patches into one pr, we need to ensure everything works when they all are combined

+1

**Comment by Adebesin-Cell** [comment]
This PR https://github.com/EveripediaNetwork/ep-ui/pull/1810/ should prolly fix it

**Comment by kesar** [comment]
> This PR #1810 should prolly fix it

then we can close this one?

**Comment by Adebesin-Cell** [comment]
> > This PR #1810 should prolly fix it
> 
> then we can close this one?

We should merge this to allow switchChain function in magic connector, but the issue with failing switch is cos of polygon mumbai deprecation

**Pull request closed without merging**